File: week5/ee/큐.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./10845.txt";

const input = fs.readFileSync(filePath).toString().trim().split("\n");

const N = Number(input[0])

let stack = []
let ans = []

for (let i = 1; i <= N; i++) {

  const cmd = input[i].trim().split(" ")

  switch (cmd[0]) {
    case 'push':
      stack.push(parseInt(cmd[1]));
      break;

    case 'pop':
      if (stack.length > 0) {
        ans.push(stack.shift())
      } else {
        ans.push(-1)
      }
      break;

    case 'size':
      ans.push(stack.length);
      break;
      
    case 'empty':
      if (stack.length > 0) {
        ans.push(0)
      } else {
        ans.push(1)
      }
      break;

    case 'front':
      if (stack.length > 0) {
        ans.push(stack.at(0))
      } else {
        ans.push(-1)
      }
      break;

    case 'back':
      if (stack.length > 0) {
        ans.push(stack.at(-1))
      } else {
        ans.push(-1)
      }
      break;

  }
}

console.log(ans.join('\n'));

File: week5/KJW/큐.py
from collections import deque
n = int(input())
q = deque()

for _ in range(n):
    data = input().split()
    if len(data) == 2:
        q.append(data[1])
    else:
        if data[0] == 'pop':
            if q:
                result = q.popleft()
                print(result)
            else:
                print(-1)
        elif data[0] == 'size':
            print(len(q))
        elif data[0] == 'empty':
            if q:
                print(0)
            else:
                print(1)
        elif data[0] == 'front':
            if q:
                print(q[0])
            else:
                print(-1)
        elif data[0] == 'back':
            if q:
                print(q[len(q)-1])
            else:
                print(-1)